### ISSUE TITLE
Not all Envoy retry policies are supported

### ISSUE THREAD

**Issue by aberres** [emissary-ingress/emissary#4383]
**To Reproduce**

Set a retry policy in a mapping like this:

```
  retry_policy:
    retry_on: "reset"
```

The CRD validation will fail with 

```Unsupported value: "reset": supported values: "5xx", "gateway-error", "connect-failure", "retriable-4xx", "refused-stream", "retriable-status-codes"```

Find the supported (new) policies here: Check the docs of supported policies at https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/router_filter#x-envoy-max-retries 

**Expected behavior**

`reset` (and maybe others)  can be used a policy.

**Comment by cindymullins-dw** [comment]
@aberres, I suppose you've had a look at the[ retry polices](https://www.getambassador.io/docs/edge-stack/latest/topics/using/retries/#retry_on) we do have available.  Can you explain your use case where the `reset` one would be needed / useful to you?

**Comment by aberres** [comment]
@cindymullins-dw Yeah, I checked those out. The list matches the options which were available in envoy 1.9 (this is also the version the docs point too. Now that envoy 1.22 is used by emissary, I think it is plausible to add the new options provided by upstream. Actually, I think they should be synced with every envoy upgrade.

In my case, I am facing intermittent `503 UC` errors. I wanted to check if it makes sense to retry these cases with the `reset` policy.
